### ISSUE TITLE
initwindow sometimes chooses interlaced video modes when progressive is available 

### ISSUE THREAD

**Issue by alanbork** [kernelkinetic/raylib#8]
on my tv InitWindow(720,480, "dots") sets the tv to 480i. a better choice would be 480p (if supported). best of all would be a flag indicating the preference of the program ie SetConfigFlags(FLAG_INTERLACED) or SetConfigFlags(FLAG_PROGRESSIVE). But short of that you should default to progressive. 

Note that sometimes interlaced is the only option - my tv supports 1080i but not 1080p. if initwindow asks for 1920,1080 and 1080i is the only resolution available, that's what should get set.

**Comment by kernelkinetic** [comment]
Would you confirm that it's enough to default to progressive if no FLAG_INTERLACED is set?

**Comment by alanbork** [comment]
yeah, that sounds reasonable. You are saying two flags is redundant, right?

**Comment by kernelkinetic** [comment]
Exactly. I would default to select a progressive format and only if the flag is set choose a interlaced. But first I need to see how to get this information 😀

**Comment by alanbork** [comment]
I think it's a matter of choosing the right mode from the array of
supported modes- some are interlaced and some are progressive.

**Comment by kernelkinetic** [comment]
Okay, half way down already. I think you have more experience @alanbork, so please help me: If interlaced flag is set but no viable interlaced mode found, should it fail because there is also no matching progressive mode? Or should it go through all available modes again and try to choose any matching mode regardless the flag?

**Comment by alanbork** [comment]
for my use case it would be better to fail so that I know it didn't
work - but I can always hack the source to get that behavior. but for
most people I think it's preferable to get some visual output even if
it's not exactly what you asked for.

**Comment by kernelkinetic** [comment]
Added in https://github.com/kernelkinetic/raylib/commit/0fe9c314deedf8c610619d3dcd8a94b23314d017, some checks are needed because I can't reproduce many different video modes here.

**Comment by alanbork** [comment]
leaving out the interlaced flag, (ie wanting progressive if possible) it works great if there is a progressive mode to use, but barfs if not. Not sure if that's what you were going for or not, but check out the wild resolution it goes for in that case...

root@raspberrypi:~/raylib4k/examples# ./dots 1920 1080
INFO: Initializing raylib 3.1-dev
INFO: DISPLAY: no graphic card set, trying card1
TRACE: DISPLAY: 1 connectors found
TRACE: DISPLAY: connector index 0
TRACE: DISPLAY: there are 11 connector modes
TRACE: DRM mode connected
TRACE: selecting for (res 1920x1080)
TRACE: mode 10 h=640 v=480
TRACE: mode 9 h=640 v=480
TRACE: mode 8 h=720 v=480
TRACE: interlaced mode
TRACE: but shouldn't choose an interlaced mode
TRACE: mode 7 h=720 v=480
TRACE: interlaced mode
TRACE: but shouldn't choose an interlaced mode
TRACE: mode 6 h=720 v=480
TRACE: mode 5 h=720 v=480
TRACE: mode 4 h=1440 v=480
TRACE: interlaced mode
TRACE: but shouldn't choose an interlaced mode
TRACE: mode 3 h=1280 v=720
TRACE: mode 2 h=1280 v=720
TRACE: mode 1 h=1920 v=1080
TRACE: interlaced mode
TRACE: but shouldn't choose an interlaced mode
TRACE: mode 0 h=1920 v=1080
TRACE: interlaced mode
TRACE: but shouldn't choose an interlaced mode
TRACE: mode -1 h=0 v=0
TRACE: mode -2 h=2 v=0
TRACE: interlaced mode
TRACE: but shouldn't choose an interlaced mode
TRACE: mode -3 h=89 v=0
TRACE: interlaced mode
TRACE: but shouldn't choose an interlaced mode
TRACE: mode -4 h=525 v=0
TRACE: mode -5 h=525 v=0
TRACE: mode -6 h=525 v=0
TRACE: mode -7 h=525 v=0
TRACE: mode -8 h=525 v=0
TRACE: mode -9 h=525 v=0
TRACE: mode -10 h=525 v=0
TRACE: mode -11 h=750 v=0
TRACE: mode -12 h=750 v=0
TRACE: mode -13 h=12849 v=0
TRACE: mode -14 h=105 v=0
TRACE: interlaced mode
TRACE: but shouldn't choose an interlaced mode
TRACE: mode -15 h=0 v=0
TRACE: mode -16 h=90 v=0
TRACE: interlaced mode
TRACE: but shouldn't choose an interlaced mode
TRACE: mode -17 h=87 v=0
TRACE: mode -18 h=4104 v=0
TRACE: interlaced mode
TRACE: but shouldn't choose an interlaced mode
TRACE: mode -19 h=0 v=0
TRACE: mode -20 h=0 v=0
TRACE: mode -21 h=0 v=0
TRACE: mode -22 h=0 v=0
TRACE: mode -23 h=0 v=0
TRACE: mode -24 h=0 v=0
TRACE: mode -25 h=0 v=0
TRACE: mode -26 h=0 v=0
TRACE: mode -27 h=0 v=0
TRACE: mode -28 h=0 v=0
TRACE: mode -29 h=0 v=0
TRACE: mode -30 h=0 v=0
TRACE: mode -31 h=0 v=0
TRACE: mode -32 h=0 v=0
TRACE: mode -33 h=10 v=0
TRACE: mode -34 h=17 v=21364
TRACE: mode -35 h=0 v=22380
TRACE: mode -36 h=25 v=17008
TRACE: interlaced mode
TRACE: but shouldn't choose an interlaced mode
TRACE: mode -37 h=26469 v=25940
TRACE: above mode selected
TRACE: DISPLAY: 32 EGL configs available
TRACE: DISPLAY: 32 matching EGL configs available
TRACE: DISPLAY: using EGL config 0
INFO: DISPLAY: Upscaling required: Screen size (1920x1080) smaller than display size (26469x25940)
INFO: DISPLAY: Device initialized successfully
INFO:     > Display size: 26469 x 25940
INFO:     > Render size:  1920 x 1882
INFO:     > Screen size:  1920 x 1080
INFO:     > Viewport offsets: 0, 802
INFO: GL: OpenGL device information:
INFO:     > Vendor:   Broadcom
INFO:     > Renderer: V3D 4.2
INFO:     > Version:  OpenGL ES 3.1 Mesa 19.3.2
INFO:     > GLSL:     OpenGL ES GLSL ES 3.10
INFO: GL: Supported extensions count: 76
INFO: GL: VAO extension detected, VAO functions initialized successfully
INFO: GL: NPOT textures extension detected, full NPOT textures supported
INFO: GL: ETC1 compressed textures supported
INFO: TEXTURE: [ID 1] Texture created successfully (1x1 - 1 mipmaps)
INFO: TEXTURE: [ID 1] Default texture loaded successfully
INFO: SHADER: [ID 1] Compiled successfully
INFO: SHADER: [ID 2] Compiled successfully
INFO: SHADER: [ID 3] Program loaded successfully
INFO: SHADER: [ID 3] Default shader loaded successfully
INFO: RLGL: Internal vertex buffers initialized successfully in RAM (CPU)
INFO: RLGL: Render batch vertex buffers loaded successfully
INFO: RLGL: Default state initialized successfully
INFO: TEXTURE: [ID 2] Texture created successfully (32x32 - 1 mipmaps)
INFO: TEXTURE: [ID 3] Texture created successfully (128x128 - 1 mipmaps)
INFO: FONT: Default font loaded successfully
WARNING: RPI: Failed to open Gamepad device, no gamepad available
WARNING: RPI: Failed to change keyboard mode (SSH keyboard?)
INFO: TEXTURE: [ID 4] Texture created successfully (1920x1080 - 1 mipmaps)
ERROR: drmModeAddFB failed with -22

**Comment by alanbork** [comment]
ps if you set FLAG_INTERLACED_HINT it works perfectly, getting interlaced when asked for and progressive if there isn't one.

assuming that's the desired behavior is there any data structure I can query to see if I got the interlaced or progressive mode requested?

**Comment by kernelkinetic** [comment]
Currently not, it is only checked during the mode selection in `InitGraphics`.

**Comment by alanbork** [comment]
in case the report above is not clear, 0fe9c31 failed when I asked for 1080p and the tv only supported 1080i.

**Comment by kernelkinetic** [comment]
Wow, what a nice resolution it found 😀 Thanks for the feedback! I expect that it just choose 1080i. I'll look into that.

**Comment by alanbork** [comment]
once you have a new build I'll try on another tv that has tons of resolutions and even different refresh rates. could be interesting!
